Source/WebCore/dom/ViewTransition.cpp

```diff
@@ -260,6 +262,32 @@ static ExceptionOr<void> checkDuplicateViewTransitionName(const AtomString& name
     return { };
 }
 
+static RefPtr<ImageBuffer> snapshotNodeVisualOverflowClippedToViewport(LocalFrame& frame, Node& node, SnapshotOptions&& options, LayoutRect& oldOverflowRect)
+{
+    if (!node.renderer())
+        return nullptr;
+
+    OptionSet<PaintBehavior> paintBehavior = frame.view()->paintBehavior();
+    Color backgroundColor = frame.view()->baseBackgroundColor();
+
+    frame.view()->setBaseBackgroundColor(Color::transparentBlack);
+    frame.view()->setNodeToDraw(&node);
+
+    ASSERT(node.renderer()->hasLayer());
+    RenderLayerModelObject* layerRenderer = downcast<RenderLayerModelObject>(node.renderer());
```

Comment:
```suggestion
    auto* layerRenderer = downcast<RenderLayerModelObject>(node.renderer());
```

(Also does this need a smart pointer?)